ui/admin/app/components/form/auth-method/index.hbs

```diff
@@ -10,6 +10,8 @@
     edit=@edit
     submit=@submit
     cancel=@cancel
+    addOption=this.addOption
+    removeOptionByIndex=this.removeOptionByIndex
```

Comment:
I just had a thought, if we're using a wrapper component that doesn't just handle `value` like currently for the two components, won't we need to have multiple functions for them?

I wonder if we should discuss trying to make the components handle these so we can remove the need for `addOption` or `removeOptions` on the consumer. I had originally only left it to the consumer because I didn't want to try and generalize handling the tracked array but it seems like we should now?

Reply:
could you elaborate on what you mean by multiple functions?

Reply:
What would it look like if we also used the key value list or the yet to be made select list wrapper? I assume they won't be the same function as the current `addOption`?

Reply:
that is a good point, passing the model to the components can help remove these functions from consumers in case of different types. something like this, let me know what you think.

```
    const field = this.args.name;
    const existingArray = this.args.model[field] ?? [];
    const newArray = [
      ...existingArray,
      { key: this.newOptionKey, value: this.newOptionValue },
    ];
    set(this.args.model, field, newArray);

    this.newOptionKey = '';
    this.newOptionValue = '';
```

Reply:
Yup I think it's worth making the change since we use nearly use it the same way everywhere (except I think the worker tags?)

We could just leave the current way in and check if we have a function to use that first, otherwise just use the model/name (maybe also leave the `@options` as optional).